apps/wolfsshd/wolfsshd.c

```diff
@@ -1882,6 +1915,13 @@ static void* HandleConnection(void* arg)
     /* check if there is a response to the shutdown */
     wolfSSH_free(ssh);
     if (conn != NULL) {
+        byte sc[1024];
+        shutdown(conn->fd, 1);
+        /* Spin until socket closes. */
+        do {
+            ret = (int)recv(conn->fd, sc, 1024, 0);
```

Comment:
Need to test for error (`ret` `-1`) here else this will get stuck in an infinite loop.